woocommerce/woocommerce-ios
# Blaze: Integrate fetching impressions on the budget setting screen

Closes: #11599 


## Description

This PR wraps up the budget settings screen by integrating the fetch for estimated impressions. Changes include:
- Updated `BlazeForecastedImpressionsInput` with `totalBudget` being Double instead of String.
- Added an enum `ForecastedImpressionState` and updated `BlazeBudgetSettingView` following the state.
- Updated `BlazeBudgetSettingViewModel` to observe changes in daily budget, start date and day count to fetch estimated impressions.
- Updated `BlazeCampaignCreationFormViewModel` to send site ID and target options to `BlazeBudgetSettingViewModel`.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented. Select the budget field.
- Since the response for forecasted impressions are mocked at the moment, so you'll see the mocked data displayed for estimated impressions.
- Feel free to update the `updateImpressions` method in `BlazeBudgetSettingViewModel` to set the state to error or loading to confirm that the UI is updated accordingly.

## Screenshots


 📝 I improvised the loading and error states. I added a comment on Figma regarding the error state, will update later when I get a response from Ann.
| Loading state | Result state | Error state |
| ----- | ----- | ----- |
| <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/d2b85025-c80e-4381-9aab-130becb9dc71" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/a356e7d6-835e-4dfa-a528-18c9b7eb5f81" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/7ed3e5ec-8907-4cea-a073-9a17a31851b0" width=320 /> | 

---
